pyrobosim/pyrobosim/core/world.py

```diff
@@ -609,6 +613,125 @@ def remove_all_objects(self, restart_numbering=True):
         if restart_numbering:
             self.object_instance_counts = {}
 
+    def add_robot(self, robot, loc=None, pose=None):
+        """
+        Adds a robot to the world given either a world entity and/or pose.
+
+        :param robot: Robot instance to add to the world.
+        :type robot: :class:`pyrobosim.core.robot.Robot`
+        :param loc: World entity instance or name to place the robot.
+        :type loc: Entity, optional
+        :param pose: Pose at which to add the robot. If not specified, will be sampled.
+        :type pose: :class:`pyrobosim.utils.pose.Pose`
+        """
+        # Check that the robot name doesn't already exist.
+        if robot.name in self.get_robot_names():
+            warnings.warn(f"Robot name {robot.name} already exists in world.")
+            return
+
+        # If the new robot has a bigger inflation radius than previously,
+        # use this new one. Otherwise, we can leave it as is.
+        old_inflation_radius = self.inflation_radius
+        new_inflation_radius = max([r.radius for r in self.robots] + [robot.radius])
```

Comment:
There are few more places with single variable.
Should we enforce the "no single letter variable" ?

Reply:
Probably a good idea, updated all the ones I found in `world.py`.

Is this enforceable in e.g. the `pre-commit-config.yaml` file or something? If so, might be worth automating in another PR.

Reply:
Should be possible for normal variables, not sure about the ones in list comprehension.